pkg/controllers/dnspolicy/dns_helper.go

```diff
@@ -268,8 +268,9 @@ func (dh *dnsHelper) setEndpoints(ctx context.Context, mcgTarget *dns.MultiClust
 	// count will track whether a new endpoint has been removed.
 	// first newEndpoints are checked based on probe status and removed if unhealthy true and the consecutive failures are greater than the threshold.
 	removedEndpoints := 0
+
 	for i := 0; i < len(newEndpoints); i++ {
-		checkProbes := getProbesForEndpoint(newEndpoints[i], probes)
+		checkProbes := getProbesForEndpoint(newEndpoints[i], probes, dnsPolicy.Name, string(listener.Name))
```

Comment:
Why use the `dnsPoilicy.Name` here? Would gateway  name not make more sense? Would also be more consistent with the DNSRecord naming see https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/controllers/dnspolicy/dns_helper.go#L402

Reply:
Yeah nice. Didn't know about that one. I've used the dnsnaming function here now. And gateway name rather than policy name. Changes are pushed.